server/endpoints/system.js

```diff
@@ -477,6 +478,45 @@ function systemEndpoints(app) {
       }
     }
   );
+
+  app.get("/system/welcome-messages", async function (request, response) {
+    try {
+      const welcomeMessages = await WelcomeMessages.getMessages();
+      response.status(200).json({ welcomeMessages });
+    } catch (error) {
+      console.error("Error fetching welcome messages:", error);
+      response.status(500).json({ message: "Internal server error" });
+    }
+  });
+
+  app.post(
+    "/system/set-welcome-messages",
+    [validatedRequest],
+    async (request, response) => {
+      try {
+        if (
+          response.locals.multiUserMode &&
+          response.locals.user?.role !== "admin"
+        ) {
+          return response.sendStatus(401).end();
+        }
+        if (!request.body || !Array.isArray(request.body.messages)) {
+          return response.status(400).json({
+            message: "Invalid message format. Expected an array of messages.",
+          });
+        }
```

Comment:
Oh this is why you are specifying the content-type in the POST.

Just use `reqBody` function  - can then restructure the intended result automatically.
`const { messages = [] } = reqBody(request);`